controllers/datadogagent/feature/prometheus_scrape/feature.go

```diff
@@ -0,0 +1,161 @@
+		if prometheusScrape.AdditionalConfigs != nil {
+			f.additionalConfigs = *prometheusScrape.AdditionalConfigs
+		}
+		reqComp = feature.RequiredComponents{
+			Agent: feature.RequiredComponent{
+				IsRequired: &f.enable,
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.CoreAgentContainerName,
+				},
+			},
+			ClusterAgent: feature.RequiredComponent{
+				IsRequired: &f.enable,
+				Containers: []apicommonv1.AgentContainerName{
+					apicommonv1.ClusterAgentContainerName,
+				},
+			},
+		}
+	}
+
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *prometheusScrapeFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *prometheusScrapeFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.ClusterAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDPrometheusScrapeEnabled,
+		Value: "true",
+	})
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.ClusterAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDPrometheusScrapeServiceEndpoints,
+		Value: strconv.FormatBool(f.enableServiceEndpoints),
+	})
+	if f.additionalConfigs != "" {
+		managers.EnvVar().AddEnvVarToContainer(apicommonv1.ClusterAgentContainerName, &corev1.EnvVar{
+			Name:  apicommon.DDPrometheusScrapeChecks,
+			Value: yamlToJSONAdditionalConfigs(f.additionalConfigs),
+		})
+	}
+
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *prometheusScrapeFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDPrometheusScrapeEnabled,
+		Value: "true",
+	})
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+		Name:  apicommon.DDPrometheusScrapeServiceEndpoints,
+		Value: strconv.FormatBool(f.enableServiceEndpoints),
+	})
+	if f.additionalConfigs != "" {
+		managers.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
+			Name:  apicommon.DDPrometheusScrapeChecks,
+			Value: yamlToJSONAdditionalConfigs(f.additionalConfigs),
+		})
+	}
+
+	return nil
+}
+
+// ManageClusterChecksRunner allows a feature to configure the ClusterChecksRunner's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *prometheusScrapeFeature) ManageClusterChecksRunner(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+func yamlToJSONAdditionalConfigs(yamlConfigs string) string {
```

Comment:
can we move this to a common file? perhaps apis/utils/utils.go?
```suggestion
func yamlToJSONString(yamlConfigs string) string {
```